File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.swift
import Foundation

func solution(_ x:Int, _ y:Int, _ z:Int) -> Void {
    let set = Set([x, y, z])
    if set.count == 1 { print("Equilateral"); return }
    var sum = x + y + z - set.max()!
    if sum <= set.max()! { print("Invalid"); return }
    if set.count == 2 { print("Isosceles"); return }
    print("Scalene")
}

while true {
    let n = readLine()!.components(separatedBy: [" "]).map { Int($0)! }
    let (a, b, c) = (n[0], n[1], n[2])
    if a == 0 && b == 0 && c == 0 { break }
    solution(a, b, c)
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int a, b, c;
    vector<int> v;
    
    while (true) {
        cin >> a >> b >> c;
        
        if (a == 0 && b == 0 && c == 0) { return 0; }
        v = {a, b, c};
        sort(v.begin(), v.end());
        
        if (v[2] >= v[0] + v[1]) {
            cout << "Invalid" << '\n';
        } else {
            if (v[0] == v[1] && v[1] == v[2]) {
                cout << "Equilateral" << '\n';
            } else if (v[0] != v[1] && v[1] != v[2] && v[0] != v[2]) {
                cout << "Scalene" << '\n';
            } else {
                cout << "Isosceles" << '\n';
            }
        }
    }
}
